### PR TITLE
Update for issue https://github.com/resteasy/resteasy-grpc/issues/33

### PR DESCRIPTION
**Pull request by ronsigal**

### CODE REVIEW

**Comment by ronsigal** [comment]
Still need to update documentation.

**Comment by ronsigal** [comment]
Ok, I've added the documentation. It reminded me that I have a question about the treatment of classnames in the com.google.protobuf.Any class, which affects the documentation. I've posed the question here: https://github.com/grpc/grpc-java/issues/10535. I wouldn't want to merge this until it's resolved.

File: grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java [hunk 1]
```diff
@@ -129,7 +130,7 @@ public HttpServletRequestImpl(final ServletResponse servletResponse, final Servl
         if ("com.google.protobuf.Any".equals(retn)) {
             acceptList = new ArrayList<String>();
             acceptList.add("true");
-            headers.put(GRPC_RETURN_RESPONSE, acceptList);
+            headers.put(ANY, acceptList);
```
Comment (jamezp) [inline grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java:133]: Is `ANY` a valid HTTP header? For http/2 header keys must be lowercase. 
Per [RFC-7540](https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2)
> However, header field names MUST be converted to lowercase prior to their encoding in HTTP/2. A request or response containing uppercase header field names MUST be treated as malformed ([Section 8.1.2.6](https://httpwg.org/specs/rfc7540.html#malformed)).

Or is this not an HTTP header?

File: grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java [hunk 2]
```diff
@@ -766,7 +767,7 @@ public void setReturnType(String returnType) {
         if ("com.google.protobuf.Any".equals(returnType) || "Any".equals(returnType)) {
             List<String> list = new ArrayList<String>();
             list.add("true");
-            headers.put(GRPC_RETURN_RESPONSE, list);
+            headers.put(ANY, list);
```
Comment (jamezp) [inline grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java:770]: Same as above.

File: grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java [hunk 3]
```diff
@@ -142,12 +144,16 @@ private void classHeader(Scanner scanner, StringBuilder sb, String fileName) {
     }
 
     private void imports(Scanner scanner, StringBuilder sb, String fileName) {
-        sb.append("import com.google.protobuf.Descriptors.FieldDescriptor;" + LS)
+        sb.append("import static dev.resteasy.grpc.bridge.runtime.Constants.ANY;" + LS)
+                .append("import com.google.protobuf.Descriptors.FieldDescriptor;" + LS)
                 .append("import com.google.protobuf.GeneratedMessageV3;" + LS)
                 .append("import com.google.protobuf.Timestamp;" + LS)
+                .append("import dev.resteasy.grpc.bridge.runtime.Utility;" + LS)
```
Comment (jamezp) [inline grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java:151]: Is this needed? I don't see any usage being added.

File: grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java [hunk 4]
```diff
@@ -142,12 +144,16 @@ private void classHeader(Scanner scanner, StringBuilder sb, String fileName) {
...
                 .append("import io.grpc.stub.StreamObserver;" + LS)
                 .append("import java.io.ByteArrayInputStream;" + LS)
                 .append("import java.io.ByteArrayOutputStream;" + LS)
+                .append("import java.io.InputStream;" + LS)
+                .append("import java.lang.reflect.Method;" + LS)
```
Comment (jamezp) [inline grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java:156]: Are these needed? Again, I don't see any usage being added.

**Review by jamezp — COMMENTED** [review]
I need to read more on how "google.protobuf.Any" works. My initial thought was can I stuff a bad actor in the payload and have it deserialized becaused of the `google.protobuf.Any` type. I could be completely wrong on this, so I just need to read up on it :)

Other than that I just have a few minor comments. Also we should probably add license headers to the new test source files.

**Comment by ronsigal** [comment]
>can I stuff a bad actor in the payload and have it deserialized becaused of the google.protobuf.Any type

Well, in general, yes, but, on the server side, *JavabufTranslator will translate only those classes it knows about.

**Comment by ronsigal** [comment]
> we should probably add license headers to the new test source files

Ah, ok.

**Comment by jamezp** [comment]
> > can I stuff a bad actor in the payload and have it deserialized becaused of the google.protobuf.Any type
> 
> Well, in general, yes, but, on the server side, *JavabufTranslator will translate only those classes it knows about.

Ah-ha, I knew I had to be missing something. Thank you!

File: grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java (hunk 1 above)
Comment (ronsigal) [reply grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java:133]: > is this not an HTTP header

Actually, I'm using it to communicate with *MessageBodyReaderWriter. I think I can remove it once *MessageBodyReaderWriter is done with it.

File: grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java (hunk 2 above)
Comment (ronsigal) [reply grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java:770]: Same response

File: grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java (hunk 3 above)
Comment (ronsigal) [reply grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java:151]: Ah, ANY used to be in another class, so it was already referenced. Lines 282 and 292. Very reasonable question, though.

File: grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java (hunk 4 above)
Comment (ronsigal) [reply grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java:156]: It looks like you're right. I see a few others that aren't used. Maybe they were used once upon a time.

File: grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java (hunk 1 above)
Comment (ronsigal) [reply grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java:133]: Removed

File: grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java (hunk 2 above)
Comment (ronsigal) [reply grpc-bridge-runtime/src/main/java/dev/resteasy/grpc/bridge/runtime/servlet/HttpServletRequestImpl.java:770]: Removed

File: grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java (hunk 4 above)
Comment (ronsigal) [reply grpc-bridge/src/main/java/dev/resteasy/grpc/bridge/generator/ServiceGrpcExtender.java]: Removed

**Comment by jamezp** [comment]
> @jamezp, I've responded to every comment except the one about licenses. I see that none of the classes in grpc-tests have such a header. Should I update all of them?

Perfect, thanks @ronsigal. This looks like it sill needs a push as I don't see that changes.

WRT the headers, we don't need to worry about that. We can do a follow up PR where we add headers on the files if we want later. We don't need to handle this in this PR.

**Review by jamezp — APPROVED** [review]

**Comment by ronsigal** [comment]
I just added some new stuff for arrays. I should have committed this pull request before I did that. D'oh.

**Comment by ronsigal** [comment]
I backed out the array related changes, so now I can merge it.

**Pull request merged**